### ISSUE TITLE
Option to remove redundant subdomains

### ISSUE THREAD

**Issue by b3t33** [hectorm/hblock#68]
The produced list of hosts contains multiple entries, whose subdomains are also included. When using DNS resolver for blocking, such as unbound, subdomains get blocked automatically and these further "more special" subdomains are redundant. Some examples are `vdo.ai` (remove all `*.vdo.ai`) or `googlesyndication.com`.

Hence, I would like you to include some filtering option, this could be done like
```
$ cat filter.sh
function reverse {
  awk -F. '{for(i=NF;i>0;i--){printf("%s%s",$i,(i>1?FS:RS))}}'
}

reverse \
  | sort -t . \
  | awk '{if (index($0,p)!=1) {print $0; p=$0".";}}' p=. \
  | reverse

$ cat hosts.txt | sh filter.sh
```
where `hosts.txt` contains all hosts/URLs. It would be best to apply the filter within your script, to have not to deal with the header and footer stuff. However, I'm no shell (and even POSIX compliant) expert...

Possibly also the resulting sort order is more natural for URLs, sorted by TLD, domain, subdomain, etc. Moreover, the size of the blocklist is roughly cut to half in terms of entries (even more in byte size).

**Comment by hectorm** [comment]
I think it would be a useful feature, I'll add it with an extra option (and disabled by default).

**Comment by b3t33** [comment]
That sounds like a plan. Much appreciated

**Comment by hectorm** [comment]
I've implemented this feature in the development branch (21b4c3d7874319a2fb90de35444dfa07169696ec). It's still experimental but I would like to have your feedback.

**Comment by b3t33** [comment]
This looks great so far! And a short test confirmed it to be working fine.

Concerning the sorting entries step afterwards, I personally really like the sort order produced in the [filtering step](https://github.com/hectorm/hblock/blob/21b4c3d7874319a2fb90de35444dfa07169696ec/hblock#L611).
By modifying this line to
```
| sort | uniq |  sed -e '/^$/d' | awk "${awkFilterScript:?}" \
```
the remaining sorting step could be dropped in the case of "filterSubdomains" (perhaps moved into if's else branch).

But it boils down to personal preference, keeping it as it is would be consistent with `hblock`'s output so far, whereas with this new option also the sort order (in that case/or in general) could be modified.

To sum up, I am very happy with the added functionality as it is. Huge thanks!

**Comment by hectorm** [comment]
I would like the order to be the same regardless of whether or not this feature is used, and in case of not using it it's faster to just use `sort`. Human readability isn't a priority for me for large domain lists.

I'm going to release a minor version. Thank you for your time to test this feature!
